### PR TITLE
add testcase for intersection prevention

### LINKED ISSUE #1154: missing testcase and doc for implementation of 'prevention of mesh intersection'
**Is your feature request related to a problem? **
PR #1076 is missing documentation and a testcase. This is the PR where we check the mesh validity after a mesh deformation.
Specifically, we check if a mesh is self-intersecting.

**Describe the solution you'd like**
A testcase explaining and showing the usage of the method (config options and description) should be added. 
A testcase is available from the thesis of L.Tol:  https://pure.tue.nl/ws/portalfiles/portal/165889356/0894988_Tol.pdf

### PR DESCRIPTION
**Pull request by bigfooted**
## Proposed Changes
added testcase to check and repair self-intersecting mesh and nonconvex cells after deformation 
 
## Related Work
fixes #1154, addition to implementation of #1076

## PR Checklist
- [x] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags, or simply --warnlevel=2 when using meson).
- [x] My contribution is commented and consistent with SU2 style.
- [x] I have added a test case that demonstrates my contribution, if necessary.
- [x] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

**Comment by pcarruscag** [comment]
Looks like the file is repeated?

**Comment by bigfooted** [comment]
> Looks like the file is repeated?

Ah, forgot to remove the copy, thanks!

File: TestCases/deformation/intersection_prevention/def_intersect.cfg [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+MESH_OUT_FILENAME= mesh_ffd_deformed.su2
+
+% Marker of the surface in which we are going apply the shape deformation
+DV_MARKER= (wall_inner_bottom, wall_inner_right, wall_inner_top)
+
+% Parameters of the shape deformation
+DV_PARAM= (MAIN_BOX, 0, 0, 0.0, 1.0); (MAIN_BOX, 1, 0, 0.0, 1.0); (MAIN_BOX, 2, 0, 0.0, 1.0); (MAIN_BOX, 3, 0, 0.0, 1.0); (MAIN_BOX, 4, 0, 0.0, 1.0); (MAIN_BOX, 5, 0, 0.0, 1.0); (MAIN_BOX, 6, 0, 0.0, 1.0); (MAIN_BOX, 7, 0, 0.0, 1.0); (MAIN_BOX, 8, 0, 0.0, 1.0); (MAIN_BOX, 9, 0, 0.0, 1.0); (MAIN_BOX, 10, 0, 0.0, 1.0); (MAIN_BOX, 0, 1, 0.0, 1.0); (MAIN_BOX, 1, 1, 0.0, 1.0); (MAIN_BOX, 2, 1, 0.0, 1.0); (MAIN_BOX, 3, 1, 0.0, 1.0); (MAIN_BOX, 4, 1, 0.0, 1.0); (MAIN_BOX, 5, 1, 0.0, 1.0); (MAIN_BOX, 6, 1, 0.0, 1.0); (MAIN_BOX, 7, 1, 0.0, 1.0); (MAIN_BOX, 8, 1, 0.0, 1.0); (MAIN_BOX, 9, 1, 0.0, 1.0); (MAIN_BOX, 10, 1, 0.0, 1.0) 
+
+% Value of the shape deformation
+DV_VALUE = 0, 0.0002, 0.0004, 0.0006, 0.0008, 0.001, 0.0012, 0.0014, 0.0016, 0.0018, 0.002, 0,-0.0002, -0.0004, -0.0006, -0.0008, -0.0010, -0.0012, -0.0014, -0.0016, -0.0018, -0.002
```
Comment (pcarruscag) [inline TestCases/deformation/intersection_prevention/def_intersect.cfg:340]: Is the prevention active for these values? If it is LGTM

**Review by pcarruscag — APPROVED** [review]

File: TestCases/deformation/intersection_prevention/def_intersect.cfg (hunk 1 above)
Comment (bigfooted) [reply TestCases/deformation/intersection_prevention/def_intersect.cfg:340]: yes, this is the case from the thesis that gave problems. Thanks!

**Pull request merged**